clang-tools-extra/clangd/ConfigCompile.cpp

```diff
@@ -572,32 +572,43 @@ struct FragmentCompiler {
 #else
     static llvm::Regex::RegexFlags Flags = llvm::Regex::NoFlags;
 #endif
-    auto Filters = std::make_shared<std::vector<llvm::Regex>>();
-    for (auto &HeaderPattern : F.IgnoreHeader) {
-      // Anchor on the right.
-      std::string AnchoredPattern = "(" + *HeaderPattern + ")$";
-      llvm::Regex CompiledRegex(AnchoredPattern, Flags);
-      std::string RegexError;
-      if (!CompiledRegex.isValid(RegexError)) {
-        diag(Warning,
-             llvm::formatv("Invalid regular expression '{0}': {1}",
-                           *HeaderPattern, RegexError)
-                 .str(),
-             HeaderPattern.Range);
-        continue;
+    std::shared_ptr<std::vector<llvm::Regex>> Filters;
+    if (!F.IgnoreHeader.empty()) {
+      Filters = std::make_shared<std::vector<llvm::Regex>>();
+      for (auto &HeaderPattern : F.IgnoreHeader) {
+        // Anchor on the right.
+        std::string AnchoredPattern = "(" + *HeaderPattern + ")$";
+        llvm::Regex CompiledRegex(AnchoredPattern, Flags);
+        std::string RegexError;
+        if (!CompiledRegex.isValid(RegexError)) {
+          diag(Warning,
+               llvm::formatv("Invalid regular expression '{0}': {1}",
+                             *HeaderPattern, RegexError)
+                   .str(),
+               HeaderPattern.Range);
+          continue;
+        }
+        Filters->push_back(std::move(CompiledRegex));
       }
-      Filters->push_back(std::move(CompiledRegex));
     }
-    if (Filters->empty())
+    std::optional<bool> AnalyzeSystemHeaders;
+    if (F.AnalyzeSystemHeaders.has_value())
+      AnalyzeSystemHeaders = **F.AnalyzeSystemHeaders;
```

Comment:
RHS here is no longer an optional

Reply:
Yes, it's intentional.
`F.AnalyzeSystemHeaders` is a `std::optional<Located<bool>>`, but the lambda only needs to capture a `std::optional<bool>`.
Lines 594-596 are equivalent to the following c++23 code:
```cpp
auto AnalyzeSystemHeaders = F.AnalyzeSystemHeaders.transform(
    [](const auto& LocatedValue) { return *LocatedValue; }
);
```

Reply:
> F.AnalyzeSystemHeaders is a std::optional<Located<bool>>

Hence RHS is just a `bool`, so I don't understand why we prefer a bool.

But now that I read the logic again it actually makes sense, not for C++23 idioms, but because we want to override the config field only if it was explicitly set, and inherit from parent config fragment otherwise. Can you add a comment about that?